### PR TITLE
[sc-188707]  Prevent posting comment when shortcut bot is enabled

### PR DESCRIPTION
**Pull request by hoshsadiq**
SC-188707

This PR prevents the comment spam when the shortcut bot is already enabled, as that already takes care of some this.
In addition, I clean up the code a little.

### CODE REVIEW

**Comment by hoshsadiq** [comment]
Because of the code clean up, this is probably best viewed with [whitespace diff disabled](https://github.com/launchdarkly-labs/ld-gh-actions-shortcut/pull/31/files?w=1)

File: set-link/entrypoint.sh [hunk 1]
```diff
@@ -84,35 +90,36 @@ echo "New title with removed ticket is '${new_title}'"
...
+"$OK" -j list_issue_comments "$GITHUB_REPOSITORY" "$number" | jq -e '. | map(select(.user?.login? == "shortcut-integration[bot]")) | .[0]' >/dev/null || SKIP_COMMENT="$?"
 
-if [[ "$story" != "" && ( "$body" != "$new_body" || "$title" != "$new_title" ) ]]; then
-    args=("title='$new_title'")
-    if [[ "$COMMENT_ONLY" != "1" && "$SKIP_LINK" != "1" ]]; then
-        args+=("body='$new_body'")
-    fi
+if [[ "$story" != "" ]] && { [[ "$body" != "$new_body" || "$title" != "$new_title" ]]; }; then
+  args=()
+  [[ "$title" != "$new_title" ]] && args+=("title='$new_title'")
```
Comment (hoshsadiq) [inline set-link/entrypoint.sh:105]: Interestingly this is still being added despite the if statement and the title/new_title being the same. Not really sure why.

Comment (cdelst) [reply set-link/entrypoint.sh:105]: Lol, the story of this script 😆

Comment (cdelst) [reply set-link/entrypoint.sh:105]: @hoshsadiq Is it possible there's an extra space in one of them? This script may be adding a space at the beginning sometimes... I was noticing that sometimes when I was fumbling with this script, but never got around to actually looking at it.

**Review by cdelst — APPROVED** [review]
My shell reviewing power is limited but this looks good to me! Thanks for doing this.

File: set-link/entrypoint.sh [hunk 2]
```diff
@@ -55,20 +62,19 @@ else
...
-if [[ "$new_body" != *"$link_url"* ]] && [[ -n "$story" ]] ; then
-    new_body="$link_url\n$new_body"
+if [[ "$new_body" != *"$link_url"* && -n "$story" ]]; then
+  new_body="$link_url\n$new_body"
 fi
 
 new_title="${title}"
 
-formatted_title=`echo "${new_title}" | cut -d "/" -f 3`
+formatted_title="$(cut <<<"${new_title}" -d "/" -f3)"
```
Comment (hoshsadiq) [inline set-link/entrypoint.sh]: @cdelst what's the reason for this bit? This is causing a title like: `sc-187720 - update aws version/service_legacy tag for applications/data-exporter` to become `[sc-187720] Data-exporter`

Comment (hoshsadiq) [reply set-link/entrypoint.sh]: Never mind, I'm guessing this is for the ones where it picks up the branch name as a title. I've addressed it.

File: set-link/entrypoint.sh (hunk 1 above)
Comment (hoshsadiq) [reply set-link/entrypoint.sh:105]: Cool, I've updated it to have address that.

**Review by ghost — APPROVED** [review]

**Pull request merged**